osf/migrations/0083_modify_noderequestaction_noderequest.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.9 on 2018-02-21 19:11
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('osf', '0082_merge_20180213_1502'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='noderequestaction',
+            name='permissions',
+            field=models.CharField(choices=[(b'admin', 'Admin'), (b'read', 'Read'), (b'write', 'Write')], default=b'read', max_length=5),
+        ),
+        migrations.AddField(
+            model_name='noderequestaction',
+            name='visible',
+            field=models.BooleanField(default=True),
+        ),
+        migrations.AlterUniqueTogether(
+            name='noderequest',
+            unique_together=set([]),
+        ),
+        migrations.RunSQL(
+            [
+                """
+                CREATE UNIQUE INDEX osf_noderequest_target_creator_non_accepted ON osf_noderequest (target_id, creator_id)
+                WHERE machine_state != 'accepted';
```

Comment:
The accepted state gets ignored in the index so that we can still have an index, but a user can have multiple requests for the same node, as long as one of them is accepted. This was done for the weird edge case where a user requests access, is accepted, is removed, and requests access again. With the old `unique_together` index, a user couldn't re-request access to the node, since the `target_id` and `creator_id` had to be unique, which didn't make sense in the case of a re-request.

Reply:
ahhh, gotcha 👍